### ISSUE TITLE
[easy] fail more gracefully when no functions folder

### ISSUE THREAD

**Issue by sw-yx** [netlify/zip-it-and-ship-it#24]
**- Do you want to request a *feature* or report a *bug*?**

feature

**- What is the current behavior?**

when a `functions` field is specified but the folder doesnt exist, we should create it, or warn and exit. instead, now we show this error:

```
Error: ENOENT: no such file or directory, scandir 'lambda'
    at createHandler (~/Netlify/netlify-dev-plugin/node_modules/@netlify/zip-it-and-ship-it/src/serve.js:61:6)
    at serveFunctions (~/Netlify/netlify-dev-plugin/node_modules/@netlify/zip-it-and-ship-it/src/serve.js:159:16)
```

**Comment by bcomnes** [comment]
Are you talking about the serve.js context?  👍 

For the library API, this is a fine error.  Consumers should not try to run this action on a missing directory, and if they try, it could be an indication of unexpected failure and the error is appropriate.

**Comment by sw-yx** [comment]
yea. imo we can capture this and show a nicer warning.

**Comment by bcomnes** [comment]
SGTM, lets just leave the api error reporting the same for now.

**Comment by ehmicky** [comment]
I agree that running `zip-it-and-ship-it` on a non-existing functions folder should throw an error. However, as @sw-yx points out, the error message should be improved.

**Comment by sw-yx** [comment]
fwiw i think we have implemented nice errors for this in netlify lambda and netlify dev

**Comment by ehmicky** [comment]
Thanks for the info @sw-yx! It might still be a good thing to provide a better error message when this happens from `zip-it-and-ship-it`? For users consuming this library directly.

**Comment by sw-yx** [comment]
yup im suggesting stealing the error msg

**Comment by ehmicky** [comment]
This is fixed as part of #48. I've also added a unit test as part of #47.
